### PR TITLE
Support validate directory for istioctl validate

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [x] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: istioctl/pkg/validate/validate.go [hunk 1]
```diff
@@ -276,25 +277,71 @@ func validateFiles(istioNamespace *string, defaultNamespace string, filenames []
...
 			reader = io.NopCloser(os.Stdin)
 		} else {
-			reader, err = os.Open(filename)
+			reader, err = os.Open(path)
+			if err != nil {
+				errs = multierror.Append(errs, fmt.Errorf("cannot read file %q: %v", path, err))
+				return
+			}
 		}
 		if err != nil {
```
Comment (bleggett) [inline istioctl/pkg/validate/validate.go:291]: This entire outdented `if` conditional from 291-293 seems pointless/redundant now? Maybe I'm missing something.

When would it ever get hit?

File: istioctl/pkg/validate/validate.go [hunk 2]
```diff
@@ -276,25 +277,71 @@ func validateFiles(istioNamespace *string, defaultNamespace string, filenames []
 	var errs, err error
```
Comment (bleggett) [inline istioctl/pkg/validate/validate.go:277]:
```suggestion
	var errs error
```

Do we really need `err` (singular) AND the global `errs` (plural) to both be global to the closure funcs?

I think we can probably drop `err` here and just keep it locally scoped to each local func - only `errs` needs to be in the outer scope AFAICT.

File: istioctl/pkg/validate/validate.go [hunk 3]
```diff
@@ -273,28 +274,71 @@ func validateFiles(istioNamespace *string, defaultNamespace string, filenames []
...
 
-	var errs, err error
+	var errs error
 	var reader io.ReadCloser
 	warningsByFilename := map[string]validation.Warning{}
-	for _, filename := range filenames {
-		if filename == "-" {
+
+	processFile := func(path string) {
+		var err error
```
Comment (bleggett) [inline istioctl/pkg/validate/validate.go:282]:
```suggestion
```

nit: Technically don't need this decl.

**Review by bleggett — APPROVED** [review]
Looks good to me.

**Comment by hanxiaop** [comment]
@GregHanson PTAL, thanks

**Review by GregHanson — COMMENTED** [review]
Can you add an example to the help text?

**Comment by hanxiaop** [comment]
@GregHanson Added in https://github.com/istio/istio/pull/44293/commits/b4abbb8491e7ff0b941fe75d7ca7864fa6d285d6, thanks!

**Review by GregHanson — APPROVED** [review]

**Pull request merged**